src/satosa/frontends/saml2.py

```diff
@@ -56,7 +64,11 @@ def hash_type_to_saml_name_id_format(hash_type):
         return NAMEID_FORMAT_TRANSIENT
     elif hash_type == UserIdHashType.persistent.name:
         return NAMEID_FORMAT_PERSISTENT
-    return NAMEID_FORMAT_PERSISTENT
+    elif hash_type == UserIdHashType.public_email:
+        return NAMEID_FORMAT_EMAILADDRESS
+    else:
+        raise SATOSAModuleError('Mapping to SAML NameID Format {} '
+                                'not implemented'.format(hash_type.name))
```

Comment:
This will result to the same error that #142 produces. `hash_type` may be `None`. 
I would suggest a tiny refactor (this goes for @jkakavas too) to fix this:

```
hash_name = getattr(hash_type, 'name', None)
if hash_name == ...
  ...
else:
  raise SATOSAModuleError('...{}'.format(hash_name)
```

Reply:
I think that NAMEID_FORMAT_UNSPECIFIED is the appropriate mapping for hash_type None

Reply:
I agree, this is very reasonable.